### ISSUE TITLE
Memory Limit

### ISSUE THREAD

**Issue by jrittvo** [MDMAchine/coreml-model-conversion-script#1]
What happens after the new "memory limit" threshold is hit?  Does the system resort to swapping, or does the conversion process halt (or crash)?

**Comment by MDMAchine** [comment]
> What happens after the new "memory limit" threshold is hit? Does the system resort to swapping, or does the conversion process halt (or crash)?

The way the memory manager works is VM based so the system is aware and will use more/combinations of VM/swap/real mem as it actively manages.

I think I left, commented out, the commands for swap settings but its best to let the system handle swap/real mem or the script would probably just kill itself in unpredictable ways.

Overall it tends to keep the memory usage lower/stable. And the system remains a bit more responsive than it otherwise would be. On a 8gb system with 100GB free HD memory, I can have it do up to 8-10 model outputs before you'd want to reboot the system.

Eventually if you do enough models in one shot it'll cache all the system memory and have nothing left for the vm/swap. Then it'll just start skipping the conversion until the limit (20). Possibly if you have a bigger HD it'll last longer? In my testing it ate up about 100GB and was routinely failing the conversions by model # 15 or so.

The 6.7 GB version seems to work best at least on a 8GB M1. I have the 80% of infinity (limit) version, but I have not tested how the scaling of % of system memory performs (it may not lol).

I havent figured out a way to purge this cache short of a reboot.

TLDR: It'll use more swap when it needs it, but will stay more stable overall. And when it runs out of HD space it'll just keep killing and/or CTD.
